### PR TITLE
ReconnectClientFactory

### PR DESCRIPTION
**Pull request by guntanis**
The current implementation of the ClientFactory doesn't support reconnects, this implements ReconnectingClientFactory and the necessary methods.

Can be implemented in Twisted like this:
    queueClient = beanstalk.twisted_client.BeanstalkClientFactory(
        reactor,
        beanstalk.twisted_client.Beanstalk(),
        generatorClass(params).worker
    )

### CODE REVIEW

File: beanstalk/twisted_client.py [hunk 1]
```diff
@@ -130,19 +131,27 @@ def rawDataReceived(self, data):
             self._current.appendleft(pending)
 
 
-class BeanstalkClientFactory(protocol.ClientFactory):
+class BeanstalkClientFactory(ReconnectingClientFactory):
```
Comment (sophacles) [inline beanstalk/twisted_client.py:134]: Looks like the ReconnectingClientFactory needs to be imported from... twisted?

File: beanstalk/twisted_client.py [hunk 2]
```diff
@@ -102,6 +102,7 @@ def lineReceived(self, line):
             res = pending.handler(line + "\r\n")
         except Exception, e:
             pending.fail(e)
+            p
```
Comment (sophacles) [inline beanstalk/twisted_client.py:105]: Typo or unfinished necessary code?

**Review by sophacles — CHANGES_REQUESTED** [review]
Hi thanks for the submission! The tests aren't passing, probably due to not having `ReconnectingClientFactory` imported.

Also, not sure what the reconnecting factory give us here, and if we need to change anything else to deal with loss of connection, can you expand on why this is an improvement a bit, or point me to resources to understand better?

**Comment by ysmolski** [comment]
@sophacles do you maintain this repo or should I archive it?

**Pull request closed without merging**